[Merged by Bors] - chore(Pointwise): Rename `Set` to `Set.Basic`

YaelDillies opened the pull request:
Move:
* `Algebra.Group.Pointwise.Set` → `Algebra.Group.Pointwise.Set.Basic`
* `Algebra.Group.Pointwise.Finset.NatCard` → `Algebra.Group.Pointwise.Set.Card`
* `Algebra.GroupWithZero.Pointwise.Set` → `Algebra.GroupWithZero.Pointwise.Set.Basic`

This is making space for a new file `Algebra.GroupWithZero.Pointwise.Set.Card` in #17231.

`Algebra.Group.Pointwise.Finset` already has this `.Basic`  naming scheme.

---
<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

For details on the "pull request lifecycle" in mathlib, please see:
https://leanprover-community.github.io/contribute/index.html

In particular, note that most reviewers will only notice your PR
if it passes the continuous integration checks.
Please ask for help on https://leanprover.zulipchat.com if needed.

To indicate co-authors, include lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Co-authored-by: Author Name <author@email.com>

If you are moving or deleting declarations, please include these lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Moves:
- Vector.* -> Mathlib.Vector.*
- ...

Deletions:
- Nat.bit1_add_bit1
- ...

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]

-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

b-mehta commented on Mathlib/Topology/Order/LeftRightNhds.lean:
```diff
@@ -3,8 +3,7 @@ Copyright (c) 2017 Johannes Hölzl. All rights reserved.
 -/
-import Mathlib.Algebra.Ring.Pointwise.Set
-import Mathlib.Order.Filter.AtTopBot.Group
+import Mathlib.Algebra.Ring.Pointwise.Set\nimport Mathlib.Order.Filter.AtTopBot.Group
```
```suggestion
import Mathlib.Algebra.Ring.Pointwise.Set.Basic
import Mathlib.Order.Filter.AtTopBot.Group
```

eric-wieser commented:
Can you be explicit in the description about precisely what the name is, and what the file it is making room for is called?

eric-wieser commented:
I sort of dispute the claim that space needs to be made here. Why can't we have:

* `Algebra/Group/Pointwise`
  * `Set.lean`
  * `Set/Card.lean`

eric-wieser commented:
Ah, I guess this ship has locally sailed with the adjacent `Finset`, so I'll let you have this for local consistency

bors d+

YaelDillies commented:
Is there any downside to renaming? Historically, every time we have wanted to add a second file similar to file `X` we've renamed `X` to `X.Basic` and added `X.NewFile`. Note that it also matches the `Pointwise.Finset` folder structure.

eric-wieser commented:
> Historically, every time we have wanted to add a second file similar to file `X` we've renamed `X` to `X.Basic` and added `X.NewFile`.

I claim we should stop doing this, it breaks downstream users for no good reason. But indeed, for this PR the other argument is enough to sway me.

YaelDillies commented:
> I claim we should stop doing this, it breaks downstream users for no good reason.

Historically, those folders basically never shrink after they are created, and it doesn't really make sense to me that we should have `.Basic` files iff the original author planned that the subject would extend to more than one file.

This organic growth of folders is one of the main attrayals of mathlib IMO. I don't see why we should get rid of it.

YaelDillies commented:
bors merge
